src/vast-client-serializer.js

```diff
@@ -0,0 +1,698 @@
+import { create } from 'xmlbuilder2'
+
+export default class VASTClientSerializer {
+
+  /**
+   * Serialize the given Vast-client parsed object into XML
+   * For detail about the data structure to pass see
+   * https://github.com/dailymotion/vast-client-js/blob/master/docs/api/class-reference.md
+   * @param {Object} parsedVast The VAST-Client parsed VAST object
+   * @returns Serialized VAST
+   */
+  serialize(parsedVast) {
+    const doc = create({ encoding: 'UTF-8' }, this.buildVast(parsedVast))
+    return doc.end({ format: 'xml' })
+  }
+
+  /**
+   * Build the root VAST node object
+   * @param {Object} parsedVast The VAST-Client parsed VAST object
+   * @returns {Object} The VAST node
+   */
+  buildVast(parsedVast) {
+    const ads = parsedVast.ads.map((ad) => {
+      return this.buildAd(ad)
+    })
+
+    const errors = parsedVast.errorURLTemplates.map((errorUrl) => {
+      return this.wrapCDATA(errorUrl)
+    })
+
+    return {
+      'VAST': {
+        '@version': parsedVast.version,
+        'Error': errors,
+        'Ad': ads,
+      }
+    }
+  }
+
+  /**
+   * Build an Ad node object
+   * @param {Object} ad An ad object
+   * @returns {Object} An ad node
+   */
+  buildAd(ad) {
+    return {
+      '@id': ad.id,
+      '@sequence': ad.sequence,
+      // '@conditionalAd': no current support in vast-client
```

Comment:
Yes, it's to make it obvious it's not implemnted because it's not supported by VAST client, once it's done we'll just need to uncomment and it'll work

Reply:
Perfect, that makes sense!